pkg/helm/actions/uninstall_release.go

```diff
@@ -25,3 +30,24 @@ func UninstallRelease(name string, conf *action.Configuration) (*release.Uninsta
 
 	return resp, nil
 }
+
+func UninstallReleaseAsync(name string, ns string, version string, conf *action.Configuration, coreClient corev1client.CoreV1Interface) (*kv1.Secret, error) {
+	client := action.NewUninstall(conf)
+	go func() {
+		resp, err := client.Run(name)
+		if err != nil || resp == nil {
+			return
+		}
+
+		ch := resp.Release.Chart
+		if ch != nil && ch.Metadata != nil && ch.Metadata.Name != "" && ch.Metadata.Version != "" {
+			metrics.HandleconsoleHelmUninstallsTotal(ch.Metadata.Name, ch.Metadata.Version)
+		}
+	}()
+	secretName := fmt.Sprintf("sh.helm.release.v1.%v.v%v", name, version)
+	secret, err := coreClient.Secrets(ns).Get(context.TODO(), secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, ErrReleaseNotFound
+	}
```

Comment:
Doesn't it make sense to check the Secret status before starting the uninstall goroutine?

Reply:
Uninstall needs to work irrespective of status so do not think that is useful.

Reply:
Also when the release is not found?

Then you can remove this check here or?